src/modules/stores/context/context.store.ts

```diff
@@ -188,6 +189,22 @@ export class ContextStore extends Store<ContextModel> {
     this.setState();
   }
 
+  getOrLoadInnovation(
+    innovationId: string,
+    context: AuthenticationModel['userContext']
+  ): Observable<ContextInnovationType> {
+    if (this.state.innovation) {
+      return of(this.state.innovation);
+    }
+
+    return this.contextService.getInnovationContextInfo(innovationId, context).pipe(
+      tap(innovation => {
+        this.setInnovation(innovation);
+        return innovation;
```

Comment:
I am unsure, but I don't think `tap` needs to return anything.

Reply:
Tested it and yeah, no return needed indeed, nice one